sklearn/ensemble/iforest.py

```diff
@@ -206,7 +219,9 @@ def predict(self, X):
 
         depths += _average_path_length(n_samples_leaf)
 
-        if not isinstance(self.max_samples, (numbers.Integral, np.integer)):
```

Comment:
When is `max_samples_` set? The docs say it is the actual number of samples, which is what we are after but I can't find it being set anywhere?